portia/config.py

```diff
@@ -844,28 +850,18 @@ def _construct_model_from_name(
             GenerativeModel: The constructed model.
 
         """
-        cache = None
-        if self.llm_redis_cache_url and validate_extras_dependencies("cache", raise_error=False):
-            from langchain_redis import RedisCache
-
-            cache = RedisCache(self.llm_redis_cache_url, ttl=CACHE_TTL_SECONDS, prefix="llm:")
-        elif self.llm_redis_cache_url:
-            logger().warning(  # pragma: no cover
-                "Not using cache as cache group is not installed. "  # pragma: no cover
-                "Install portia-sdk-python[caching] to use caching."  # pragma: no cover
-            )  # pragma: no cover
         match llm_provider:
             case LLMProvider.OPENAI:
                 return OpenAIGenerativeModel(
                     model_name=model_name,
                     api_key=self.must_get_api_key("openai_api_key"),
-                    cache=cache,
+                    **MODEL_EXTRA_KWARGS.get(f"openai/{model_name}", {}),
```

Comment:
Could use `llm_provider.value` here to avoid re-specifying the string:

```suggestion
                    **MODEL_EXTRA_KWARGS.get(f"{llm_provider.value}/{model_name}", {}),
```